iceoryx_posh/include/iceoryx_posh/capro/service_description.hpp

```diff
@@ -99,27 +92,18 @@ class ServiceDescription
     };
 
     /// @brief construction of the capro service description using serialized strings
-    ServiceDescription(const cxx::Serialization& f_serial) noexcept;
+    ServiceDescription(const cxx::Serialization& serial) noexcept;
 
     /// @brief default C'tor
     ServiceDescription() noexcept;
-    ServiceDescription(const ServiceDescription&) = default;
-    ServiceDescription(ServiceDescription&&) = default;
-    ~ServiceDescription() = default;
-
-    /// @brief construction of the capro service description using integers to create a service service description
-    ServiceDescription(uint16_t f_serviceID, uint16_t f_instanceID) noexcept;
-
-    /// @brief construction of the capro service description using fixed strings to create a service service description
-    ServiceDescription(const IdString_t& f_service, const IdString_t& f_instance) noexcept;
-
-    /// @brief construction of the capro service description using integers to create an event service description
-    ServiceDescription(uint16_t f_serviceID, uint16_t f_eventID, uint16_t f_instanceID) noexcept;
+    ServiceDescription(const ServiceDescription&) noexcept = default;
+    ServiceDescription(ServiceDescription&&) noexcept = default;
+    ~ServiceDescription() noexcept = default;
 
     /// @brief construction of the capro service description using fixed strings to create an event service description
-    ServiceDescription(const IdString_t& f_service,
-                       const IdString_t& f_instance,
-                       const IdString_t& f_event,
+    ServiceDescription(const IdString_t& service,
```

Comment:
I would love to avoid `isValid()` and invalid `ServiceDescription`'s. However, forbidding the user to provide an empty string is cumbersome and I only came up with those alternatives:
1. Let a factory create all `ServiceDescription`'s, decided against as the user API would get way too complicated
2. Calling the `errorHandler()` if the users passes `{"", "", ""}`, not user friendly, either
3. `static_assert(!service.empty()  &&... )` AFAIK not possible with C++14